fgpyo/util/metric.py

```diff
@@ -175,7 +175,17 @@ def parse(cls, fields: List[str]) -> Any:
         parsers = cls._parsers()
         header = cls.header()
         assert len(fields) == len(header)
-        return inspect.attr_from(cls=cls, kwargs=dict(zip(header, fields)), parsers=parsers)
+        return inspect.attr_from(
+            cls=cls, kwargs=dict(zip(header, map(cls.parser_formatting, fields))), parsers=parsers
+        )
+
+    @classmethod
+    def parser_formatting(self, field: Optional[str]) -> Any:
```

Comment:
1. when is `field` ever a `None`?  In it's usage in `parse`, the `fields passed in are of type `str`.
2. do we want to expose this function as part of the public API (i.e. a public method in `Metric`).  I'd argue to keep it private for now.